src/main/java/io/strimzi/kafka/quotas/StaticQuotaCallback.java

```diff
@@ -131,17 +156,46 @@ public void configure(Map<String, ?> configs) {
         storageQuotaHard = config.getHardStorageQuota();
         storageCheckInterval = config.getStorageCheckInterval();
         logDirs = Arrays.stream(config.getLogDirs().split(",")).map(Paths::get).collect(Collectors.toList());
+        thresholdMap = new HashMap<>();
```

Comment:
You should compute the threshold map using a local var and then publish to `this.thresholdMap` once the map is constructed. Otherwise the checked could observe it being empty. It's unlikely to cause a real bug, because it's likely the checker will run again before a threshold is exceeded, but still... Alternatively this will be fixed once #11 is fixed.

Reply:
So it's not 100% that `configure` will run (be invoked) before `quotaLimit` ?

Reply:
I was thinking that the quota callback was `Reconfigurable`, but it's not. 

But, being pedantic, `configure` will be called from a different thread than the callback is executed from, and so there is no _happens-before_ guaranteed by the Java memory model.

Reply:
Hmmm, same can then happen with logDirs ... and if they are null when the checker `run` is invoked, it will never loop.
How to handle that?

Reply:
Yeah, it's only a concern in theory. In practice it will work. Let's just try to simplfy all of this in #11 which will simplify and make it easier to test.